op-node/rollup/async/asyncgossiper_test.go

```diff
@@ -0,0 +1,140 @@
+package async
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+type mockNetwork struct {
+	reqs []*eth.ExecutionPayload
+}
+
+func (m *mockNetwork) PublishL2Payload(ctx context.Context, payload *eth.ExecutionPayload) error {
+	m.reqs = append(m.reqs, payload)
+	return nil
+}
+
+type mockMetrics struct{}
+
+func (m *mockMetrics) RecordPublishingError() {}
+
+// TestAsyncGossiper tests the AsyncGossiper component
+// because the component is small and simple, it is tested as a whole
+// this test starts, runs, clears and stops the AsyncGossiper
+// because the AsyncGossiper is run in an async component, it is tested with eventually
+func TestAsyncGossiper(t *testing.T) {
+	m := &mockNetwork{}
+	// Create a new instance of AsyncGossiper
+	p := NewAsyncGossiper(context.Background(), m, log.New(), &mockMetrics{})
+
+	// Start the AsyncGossiper
+	p.Start()
+
+	// Test that the AsyncGossiper is running within a short duration
+	require.Eventually(t, func() bool {
+		return p.running.Load()
+	}, time.Second, 10*time.Millisecond)
```

Comment:
All these timeouts need to be much longer or we'll get flakiness in CI because thread scheduling and CPU availability is really very volatile in CI.  More on the order of 10 seconds.

Reply:
Haha, I had 100ms at one point and Joshua suggested 1s. 10s it shall be!